File: src/main/java/com/hazelcast/elasticmemory/util/MemoryUnit.java
package com.hazelcast.elasticmemory.util;


public enum MemoryUnit {
	Bytes {
		public int convert(int value, MemoryUnit m) {return m.toBytes(value);}
		public int toBytes(int value) {return value;}
		public int toKiloBytes(int value) {return value/K;}
		public int toMegaBytes(int value) {return value/M;}
		public int toGigaBytes(int value) {return value/G;}
	},
	KiloBytes {
		public int convert(int value, MemoryUnit m) {return m.toKiloBytes(value);}
		public int toBytes(int value) {return value*K;}
		public int toKiloBytes(int value) {return value;}
		public int toMegaBytes(int value) {return value/K;}
		public int toGigaBytes(int value) {return value/M;}
	},
	MegaBytes {
		public int convert(int value, MemoryUnit m) {return m.toMegaBytes(value);}
		public int toBytes(int value) {return value*M;}
		public int toKiloBytes(int value) {return value*K;}
		public int toMegaBytes(int value) {return value;}
		public int toGigaBytes(int value) {return value/K;}
	},
	GigaBytes {
		public int convert(int value, MemoryUnit m) {return m.toGigaBytes(value);}
		public int toBytes(int value) {return value*G;}
		public int toKiloBytes(int value) {return value*M;}
		public int toMegaBytes(int value) {return value*K;}
		public int toGigaBytes(int value) {return value;}
	};

	private static final int K = 1024;
	private static final int M = K*K;
	private static final int G = K*M;
	
	public abstract int convert(int value, MemoryUnit m);
	public abstract int toBytes(int value);
	public abstract int toKiloBytes(int value);
	public abstract int toMegaBytes(int value);
	public abstract int toGigaBytes(int value);
}

File: src/Hazelcast.Net/Models/MemoryUnit.cs
// Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

namespace Hazelcast.Models;

/// <summary>
/// Represents a memory size unit.
/// </summary>
internal enum MemoryUnit
{
    /// <summary>
    /// Memory size in bytes.
    /// </summary>
    Bytes = 0,

    /// <summary>
    /// Memory size in kilobytes.
    /// </summary>
    KiloBytes = 1,

    /// <summary>
    /// Memory size in megabytes.
    /// </summary>
    MegaBytes = 2,

    /// <summary>
    /// MemorySize in gigabytes.
    /// </summary>
    GigaBytes = 3
}
